File: Dec2020/5Dec/WaterBottles.cpp
/*
Problem name: 1518. Water Bottles
Problem link:https://leetcode.com/problems/water-bottles/
*/

class Solution {
public:
    int numWaterBottles(int numBottles, int numExchange) {
        int drank = numBottles;
        int left;
        do{
            left = numBottles % numExchange;
            numBottles /= numExchange;
            drank += numBottles;
            numBottles += left;
        }while(numBottles >= numExchange);
        
        return drank;
    }
};

File: Dec2020/5Dec/WaterBottles.java
/*
Problem Name: 1518. Water Bottles
Problem Link: https://leetcode.com/problems/water-bottles/
*/
class Solution {
    public int numWaterBottles(int numBottles, int numExchange) {
        int sum = 0, empty = 0;
        for(;numBottles>0;){
            empty += numBottles;
            sum += numBottles;
            numBottles = empty/numExchange;
            empty %= numExchange;
        }
        return sum;
    }
}
